plugins/circleci/src/circleci-config.ts

```diff
@@ -164,17 +171,23 @@ export default abstract class CircleCiConfigHook extends Hook {
       throw error
     }
 
-    const job = {
-      [this.job]: {
-        'node-version': getOptions('@dotcom-tool-kit/circleci')?.nodeVersion ?? '16.14-browsers',
-        ...this.jobOptions
-      }
+    const computedJob = this.getComputedJob()
+
+    const hasJob = (jobs: NonNullable<Workflow['jobs']>): boolean => {
+      return !jobs.some((candidateJob, index) => { 
+                if(Object.keys(candidateJob)[0] === this.job && !isEqual(candidateJob, computedJob)) {
+                  jobs[index] = computedJob; return true
```

Comment:
**nitpick**: I find it a bit surprising that a function called `hasJob` has side effects, could we move the assignment out of this function?